### PR TITLE
[SCB-1039]Add interface to compatible with ServiceCenter Aggregator

### PR DESCRIPTION
**Pull request by heyile**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/RegistryUtils.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ public static boolean updateInstanceProperties(Map<String, String> instancePrope
   }
 
   public static Microservice getMicroservice(String microserviceId) {
-    return serviceRegistry.getRemoteMicroservice(microserviceId);
+    return serviceRegistry.getAggregatedRemoteMicroervice(microserviceId);
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/RegistryUtils.java]: not change this

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/ServiceRegistry.java [hunk 2]
```diff
@@ -58,6 +58,15 @@ MicroserviceInstances findServiceInstances(String appId, String microserviceName
 
   Microservice getRemoteMicroservice(String microserviceId);
 
+  /**
+   * <p>
+   *    if connect to simple ServiceCenter, same with the method
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/ServiceRegistry.java]: normal, not "sample"?

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/LocalServiceRegistryClientImpl.java [hunk 3]
```diff
@@ -353,6 +358,16 @@ public String getSchema(String microserviceId, String schemaId) {
     return microservice.getSchemaMap().get(schemaId);
   }
 
+  @Override
+  public String getAggregatedSchema(String microserviceId, String schemaId) {
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/LocalServiceRegistryClientImpl.java]: do not copy code
just invoke getSchema

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/LocalServiceRegistryClientImpl.java [hunk 4]
```diff
@@ -188,6 +188,11 @@ public Microservice getMicroservice(String microserviceId) {
     return microserviceIdMap.get(microserviceId);
   }
 
+  @Override
+  public Microservice getAggregatedMicroservice(String microserviceId) {
+    return microserviceIdMap.get(microserviceId);
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/LocalServiceRegistryClientImpl.java]: do not copy code
just invoke getMicroservice

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 5]
```diff
@@ -356,6 +356,30 @@ public String getSchema(String microserviceId, String schemaId) {
...
+    if (holder.value != null) {
+      return holder.value.getSchema();
+    }
+
+    return null;
+  }
+
+  @Override
+  public String getAggregatedSchema(String microserviceId, String schemaId) {
+    Holder<GetSchemaResponse> holder = new Holder<>();
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: do not copy code
just make global to be a parameter

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 6]
```diff
@@ -489,7 +534,7 @@ public String registerMicroserviceInstance(MicroserviceInstance instance) {
     CountDownLatch countDownLatch = new CountDownLatch(1);
     RestUtils.get(ipPort,
         String.format(Const.REGISTRY_API.MICROSERVICE_INSTANCE_OPERATION_ALL, providerId),
-        new RequestParam().addHeader("X-ConsumerId", consumerId),
+        new RequestParam().addHeader("X-ConsumerId", consumerId).addQueryParam("global", "true"),
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: no need to add global

Comment (heyile) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: ok

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/RegistryUtils.java (hunk 1 above)
Comment (heyile) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/RegistryUtils.java]: if change this... the method org.apache.servicecomb.serviceregistry#Microservice getAggregatedRemoteMicrservice(String microserviceId) will never be invoked? can I delete the method...

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 7]
```diff
@@ -795,7 +841,7 @@ public ServiceCenterInfo getServiceCenterInfo() {
     CountDownLatch countDownLatch = new CountDownLatch(1);
     RestUtils.get(ipPort,
         Const.REGISTRY_API.SERVICECENTER_VERSION,
-        new RequestParam(),
+        new RequestParam().addQueryParam("global", "true"),
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: no need to add global

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 8]
```diff
@@ -346,19 +346,26 @@ public boolean registerSchema(String microserviceId, String schemaId, String sch
 
   @Override
   public String getSchema(String microserviceId, String schemaId) {
+    return doGetSchema(microserviceId, schemaId, false);
+  }
+
+  private String doGetSchema(String microserviceId, String schemaId, boolean isAggregatedServiceCenter) {
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: not "isAggregatedServiceCenter"
target is ServiceCenter Aggregator:
  1.for register flow, global=false
  2.for find instance flow, global=true
so the variable name better to be: global

Comment (heyile) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: ok, i will change it

**Comment by liubao68** [comment]
Why we need to distinguish from register & find flow? I think service center knows the request is register or find.

**Comment by heyile** [comment]
> Why we need to distinguish from register & find flow? I think service center knows the request is register or find.

@wujimin  I know why, but I do not know the details very clearly

**Comment by heyile** [comment]
> Why we need to distinguish from register & find flow? I think service center knows the request is register or find.

@wujimin I do not know the details very clearly

**Comment by wujimin** [comment]
both register and find instance flow, need to invoke getMicroservice/getSchema
but:
  only local serviceCenter cluster available for register flow
  all serviceCenter clusters in aggregator available for find instance flow

ServiceCenter can not know the scenes, so need client  invoke with the flag

**Pull request merged**